### DISCUSSION TITLE
AM out there...

### DISCUSSION THREAD

**Discussion by zen0bit** [ivan-hc/AM#141] [Show and tell]
If you found some mention about AM/AppMan anywhere share link.

Or even better, if you share something, post it also here...

**Comment by zen0bit** [comment]
AM support in neofetch/hyfetch [PR](https://github.com/hykilpikonna/hyfetch/pull/234)

**Reply by zen0bit** [reply]
Merged

**Reply by Samueru-sama** [reply]
Isn't that a very slow way to get the packages? I checked the commit and it is doing a `am -f | grep 'PROGRAMS MANAGED BY' | cut -d' ' -f4` to get the list of packages and that takes a few seconds on my pc. 

Unless there is some caching going on it this is very slow.

**Reply by zen0bit** [reply]
Currently no better way I think. But i will speak with developer...

**Reply by Samueru-sama** [reply]
I will take a look at that tomorrow, the issue is that `am -f` also does other stuff like calculating the size of the packages which takes time. 

It should be as easy as just copying what the files modules does to only get the number of packages and implement that.

**Reply by ivan-hc** [reply]
I'll try to made it faster, later... maybe by implementing a --less poption to show less things, or --more to show what we see now. You choose.

**Reply by ivan-hc** [reply]
however, to play with the `du` command would be enough... for example by checking /opt/am/ * /remove. In the previous releases "am -f" worked like this... then I was gone crazy and then I decided to show what sh*t we have installed :rofl:

**Reply by Samueru-sama** [reply]
I just did a quick test by hacking the `files.am` module with just this: 

```
#!/bin/sh

cd "$APPSPATH" &&
find . -type f -name 'remove' 2>/dev/null | sed -r 's|/[^/]+$||' | sort | uniq | wc -l
```
Which prints the number of packages which is what is needed.

And it is very fast: 
```
~/ time am -f                                                                        
16
am -f  0.01s user 0.02s system 131% cpu 0.025 total
```

**Reply by ivan-hc** [reply]
I've quickest solutions in the old releases, maybe you should check there... before version 5 AM was not modular but just one script, you should check there the option... maybe also the version 4, I don't remember when I've improved -f

**Comment by ivan-hc** [comment]
hi folks, use the command `am -f --less` to see only the number of apps installed

**Reply by zen0bit** [reply]
Going to change it in hyfetch

**Reply by zen0bit** [reply]
[PR](https://github.com/hykilpikonna/hyfetch/pull/250)

**Reply by Samueru-sama** [reply]
Btw when are we getting the next release that supports `am -f --less`? I want to ask fastfetch to add AM to the list of package managers after that.

**Reply by zen0bit** [reply]
Already out there
Just update am itself. It will update modules...

**Reply by Samueru-sama** [reply]
Oh, I thought that the modules only get updated with each release lol.

**Reply by ivan-hc** [reply]
It's getting as complicated as the controls of the much acclaimed "Flatpak" XD

- `am -u` : update apps, modules, the CLI and everything
- `am -u --apps` : Update apps only
- `am -s` : updates CLI and modules and notifies changes to scripts of installed apps

**Comment by zen0bit** [comment]
> [PR](https://github.com/hykilpikonna/hyfetch/pull/250)
Merged

**Comment by ivan-hc** [comment]
hey GUIys :rofl: would you like to see what I've found on the antix forum? An ui for AppMan

https://www.antixforum.com/forums/topic/appman-a-package-manager-for-appimage-and-binary-files/

download from here https://www.antixforum.com/wp-content/uploads/2023/06/appman-gui-5.1.zip

this is genial in my opinion

**Reply by ivan-hc** [reply]
just remembered that we have an opened issue pinnud from two years, here

**Reply by Samueru-sama** [reply]
Looks good: 
![image](https://github.com/ivan-hc/AM/assets/36420837/e59f8abd-2b7f-4192-b3f2-47587dd3e159)

However, I had to move appman from `~/.local/bin` back to where the appman modules are installed for it to work.
